index.js

```diff
@@ -395,229 +402,327 @@ async function removeSharedPartFromReconciliation(
-                `${test.rollforwards.length} rollforward expectation${
-                  test.rollforwards.length > 1 ? "s" : ""
-                } failed`
-              )
-            );
-            test.rollforwards.forEach((expectation) => {
-              console.log(`At line number ${expectation.lineNumber}`);
-              console.log(
-                `For rollforward ${chalk.blue.bold(
-                  expectation.rollforwardName
-                )} got ${chalk.blue.bold(expectation.outcome)} (${chalk.italic(
-                  typeof expectation.outcome
-                )}) but expected ${chalk.blue.bold(
-                  expectation.expected
-                )} (${chalk.italic(typeof expectation.expected)})`
-              );
-            });
-            console.log("");
+          // Rollforwards
+          if (Object.keys(testElements.rollforwards).length > 0) {
+            listErrors(testElements.rollforwards, "rollforward");
           }
         });
+        break;
+      }
+  }
+}
+
+// Path to store HTML exports
+function resolveHTMLPath(testName) {
+  const homedir = require("os").homedir();
+  const folderPath = path.resolve(homedir, ".silverfin/html_exports");
+  const filePath = path.resolve(folderPath, `${testName}.html`);
+  fsUtils.createFolder(folderPath);
+  return filePath;
+}
+
+// Retrieve HTML, store it and open it in the default browser if needed
+async function getHTML(url, testName, openBrowser = false) {
+  const filePath = resolveHTMLPath(testName);
+  const htmlResponse = await axios.get(url);
+  if (htmlResponse.status === 200) {
+    fs.writeFileSync(filePath, htmlResponse.data);
+    if (openBrowser) {
+      if (isWsl) {
+        if (commandExistsSync("wsl-open")) {
+          exec(`wsl-open ${filePath}`);
+        } else {
+          console.log(
+            "In order to automatically open HTML files on WSL, we need to install the wsl-open script."
+          );
+          console.log(
+            "You might be prompted for your password in order for us to install 'sudo npm install -g wsl-open'"
+          );
+          execSync("sudo npm install -g wsl-open");
+          console.log("Installed wsl-open script");
+          exec(`wsl-open ${filePath}`);
+        }
+      } else {
+        await open(filePath);
       }
     }
+  }
+}
+
+async function deleteExistingHTMLs() {
+  try {
+    const homedir = require("os").homedir();
+    const folderPath = path.resolve(homedir, ".silverfin/html_exports");
+    if (!fs.existsSync(folderPath)) {
+      return;
+    }
+    const files = fs.readdirSync(folderPath);
+    files.forEach((fileName) => {
+      const filePath = path.resolve(folderPath, fileName);
+      fs.unlinkSync(filePath);
+    });
+  } catch (err) {}
+}
+
+async function handleHTMLfiles(testName = "", testRun) {
+  deleteExistingHTMLs();
+  if (testName) {
+    // Only one test
+    await getHTML(testRun.tests[testName].html, testName, true);
+  } else {
+    // All tests
```

Comment:
Should we make it not open all tests by default in the browser if there's more than one defined? (Add a new flag like `--open-all` / `-o`)

Reply:
I would say if you pass the `--html` flag, you would like to open them in the browser. If not, if you are not storing them anywhere so there is no point of using `--html`?

Reply:
Idd, we can leave it as is